### ISSUE TITLE
Output shape argument description in tf.keras.layers.MultiHeadAttention possibly wrong.

### ISSUE THREAD

**Issue by lcs-crr** [keras-team/tf-keras#282]
Please go to TF Forum for help and support:

https://discuss.tensorflow.org/tag/keras

If you open a GitHub issue, here is our policy:

It must be a bug, a feature request, or a significant problem with the documentation (for small docs fixes please send a PR instead).
The form below must be filled out.

**Here's why we have that policy:**.

Keras developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

**URL(s) with the issue:**.

[Link to MultiHeadAttention docs: https://www.tensorflow.org/api_docs/python/tf/keras/layers/MultiHeadAttention/
](https://www.tensorflow.org/api_docs/python/tf/keras/layers/MultiHeadAttention)

**Description of issue (what needs to be changed):**.
Output shape argument description currently says the following: _The expected shape of an output tensor, besides the batch and sequence dims. If not specified, projects back to the **key** feature dim._ I believe it should say:  _The expected shape of an output tensor, besides the batch and sequence dims. If not specified, projects back to the **input** feature dim._  

**Correct links**.
Yes

**Parameters defined**.

Yes

**Returns defined**.

Yes

**Raises listed and defined**

Yes

**Usage example**

This can be confirmed with the following code:
```
import tensorflow as tf
mha = tf.keras.layers.MultiHeadAttention(key_dim=2, num_heads=8)
input = tf.keras.Input(shape=[256, 18])
attention = mha(input, input, input)
print(mha.weights[6].shape[-1])
```
The code will output 18 (the input feature dimension) rather than the key feature dimension (in this case, 2).

**Request visuals, if applicable**.

No and no.

**Submit a pull request?**.

Yes
